### PR TITLE
XWIKI-22368: UIExtensionClass imported properties removed after editing and rolling back

### PR DESCRIPTION
**Pull request by surli**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-22368

# Changes

## Description

* Provide a test to check that mandatory document initializers are properly declared for the demo package

This PR is only about the test to add. I'm not sure it deserves to be added in other branches even if the actual fix of the ticket was backported.
Note that I put it in flavor-test-ui and not flavor-test-misc because that one apparently doesn't support yet using AbstractTest which is needed to use PO. 

## Clarifications

# Screenshots & Video

N/A

# Executed Tests

`mvn clean install -Dpattern=XClassDeclaredTest` in `xwiki-platform-distribution-flavor-test-ui` but note that right now it exposes a few problems.

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * None (check the description)

### CODE REVIEW

File: xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        }
+        {{/groovy}}        
+    """;
+
+    @Test
+    public void testXClassAreDeclaredInPackage()
+    {
+        getUtil().loginAsAdmin();
+        ViewPage page = getUtil().createPage(new DocumentReference("xwiki", "Test", "TestXClass"), GROOVY_CODE);
```
Comment (tmortagne) [inline xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java:92]: You might want to take a look at #executeWiki, which is a shotcut for this use case.

Comment (surli) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java:92]: right I forgot that one

File: xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+{
+    private static final String GROOVY_CODE =
+    """
+        {{groovy}}
+        import com.xpn.xwiki.doc.MandatoryDocumentInitializer;
+        import java.time.temporal.ChronoUnit;
+        import java.text.SimpleDateFormat;
+        
+        // define the pages that shouldn't be created during the build.
+        def allowedExceptions = ["XWiki.XWikiServerXwiki"];
```
Comment (tmortagne) [inline xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java:48]: I guess this list should ideally be a Maven/system property.

Comment (surli) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java]: I won't go there for now.

**Pull request merged**